src/Temporalio/Bridge/ByteArray.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Google.Protobuf;
+
+namespace Temporalio.Bridge
+{
+    internal class ByteArray : SafeHandle
```

Comment:
It took a long time to learn how to safely use some of these pointers in .NET. Close scrutiny on this `Temporalio.Bridge` namespace as a whole is welcome.

Reply:
Can you expand a bit on the choice of `SafeHandle` here and what that's providing (that'd be the kind of thing that'd be good in a docstring somewhere).

I think this makes good sense, but the .NET docs kinda imply it's largely about windows stuff. Just a sentence or two about how it can be used to deal with externally managed resources generally would be great

Reply:
Yeah most of the olds docs say it's for win32 handles, but it provides benefits to all unmanaged IntPtr. There was also https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.handleref which a note says is now SafeHandle. I could just implement most of https://referencesource.microsoft.com/#mscorlib/system/runtime/interopservices/safehandle.cs myself, though no value.

I will add docs to this one and I suppose tell other classes to read this class's docs. Or add a README at the root of this dir or something.

Reply:
Added minor docs